staging/src/k8s.io/apiextensions-apiserver/examples/client-go/hack/update-codegen.sh

```diff
@@ -21,6 +21,9 @@ set -o pipefail
 SCRIPT_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
 CODEGEN_PKG=${CODEGEN_PKG:-$(cd "${SCRIPT_ROOT}"; ls -d -1 ./vendor/k8s.io/code-generator 2>/dev/null || echo ../code-generator)}
 
+# go mod vendor does not set scripts as executable
+chmod +x "${CODEGEN_PKG}"/generate-groups.sh
```

Comment:
assuming this directory is writeable seems strange to me... I thought there were other GOPATH-based assumptions in the generation code... is this the only thing keeping it from working in a go modules context?

Reply:
Seems like it. 

For me, giving executable permission to vendor/k8s.io/code-generator/generate-groups.sh and vendor/k8s.io/code-generator/generate-internal-groups.sh made ./hack/update-codegen.sh to complete successfully.

Reply:
> assuming this directory is writeable seems strange to me

`$GOPATH/pkg/mod` and the dependencies downloaded in the local go mod cache after running `go mod vendor` are not writeable (https://github.com/golang/go/issues/27161, https://github.com/golang/go/issues/27455), but the `vendor` directory generated by `go mod vendor` is.

 >  thought there were other GOPATH-based assumptions in the generation code

Yes, there are. Ref: https://github.com/kubernetes/gengo/pull/144#issuecomment-480477427

>  is this the only thing keeping it from working in a go modules context?

...yes and no. This is the only thing keeping it from working in a go modules context _which follows GOPATH-based assumptions_ (but doesn't have code-generator at `$GOPATH/src/k8s.io/code-generator`). This will _not_ work for go modules which _don't_ follow GOPATH-based assumptions.